### DISCUSSION TITLE
Listen to a channel from any back end file?

### DISCUSSION THREAD

**Discussion by respectabiggle** [socketio/socket.io#4960] [Q&A] [answered]
Hello.

I'm hoping to emit from one back end file, and listen to the channel at a different back end file.  No front end at all.

Emitting file "emit.js"

> import { createServer }	from 'http'
> const httpServer = createServer()
> httpServer.listen(3000)
> import { Server } from 'socket.io'
> const io = new Server(httpServer)
>
> import { WebSocket } from 'ws'
> const ws = new WebSocket('wss://ws.coincap.io/prices?assets=ethereum')
>
>
> 
> 
> export const emit = async function (){
> 
>	ws.on('message', (event) => {
> 
>	let data = JSON.parse(event)
>	console.log(data) 					
>                 
>	io.emit('test', data)
> 
>	}) 
> 
> }

Listening file "listen.js"

> export const listen = async function (){
> 
> 	// await io.on('test', (event) => {
> 
> 	//	let data = JSON.parse(event)
> 	// 	console.log(data) 							
> 		
> 	// }) 
> 
> }
> 

What is the simple way to set this up?  Thank you so much!!!!

**Comment by darrachequesne** [comment, marked as answer]
Are those two files run in the same Node.js process? In that case, you could use a simple `EventEmitter`:

`emit.js`

```js
import { WebSocket } from 'ws'
import { EventEmitter } from 'node:events';

const ws = new WebSocket('wss://ws.coincap.io/prices?assets=ethereum')

export const myEventBus = new EventEmitter();

ws.on('message', (event) => {
  let data = JSON.parse(event)

  myEventBus.emit('test', data)
})
```

`listen.js`

```js
import { myEventBus } from './emit';

myEventBus.on('test', (data) => {
  // ...
});
```

Reference: https://nodejs.org/api/events.html

If you have two distinct Node.js processes, then you can indeed create a Socket.IO connection:

```js
import { io } from 'socket.io-client'

export const listen = async function (){

  const socket = io('http://localhost:3000');
  
  socket.on('test', (data) => {
    // ...
  })
}
```

**Reply by respectabiggle** [reply]
Thank you for telling me about EventEmitter !
It works and is awesome.

For two distinct Node processes, should I start with https://nodejs.org/api/cluster.html ?

Also, should I understand from your answer that there is not a native function in Socket.io 
to broadcast to the rest of the back end, 
within one single Node process?

Thank you again for being friendly!
